File: src/main/java/br/ufc/insightlab/vonqbe/entity/WebResultItem.java
package br.ufc.insightlab.vonqbe.entity;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;
import org.apache.jena.query.QuerySolution;

import br.ufc.insightlab.ror.entities.ResultQuery;

public class WebResultItem implements Serializable{
	
	public Map<String,String> values;
	
	public WebResultItem(ResultQuery r) {
		values = new HashMap<>();
		for(String k : r.getProjections()){
			values.put(k, r.getValue(k));
		}
	}

	public WebResultItem(QuerySolution r) {
		values = new HashMap<>();
		Iterator<String> varsIterator = r.varNames();
		while(varsIterator.hasNext()){
			String key = varsIterator.next();
			values.put(key, r.get(key).toString());
		}
	}
}

File: app/br/ufc/insightlab/vonqbe/entity/WebResultItem.scala
package br.ufc.insightlab.vonqbe.entity

import br.ufc.insightlab.ror.entities.ResultQuery
import org.apache.jena.query.QuerySolution

class WebResultItem {

  // r is of generic type Q and may be only of the type ResultQuery or QuerySoluction
  def this(r : Any) {
    this()

    r match {
      case ResultQuery => r.asInstanceOf[ResultQuery]
        .getProjections
        .map((p : String) => p -> r.asInstanceOf[ResultQuery].getValue(p))
        .toMap
        .asScala

      case QuerySolution => r.asInstanceOf[QuerySolution]
        .varNames()
        .map((p : String) => p -> r.asInstanceOf[QuerySolution].getValue(p))
        .toMap
        .asScala

      case _ => ???
    }
  }

//    //if (r instanceof ResultQuery){
//    if (r match ResultQuery){
//
//      this (
//        r
//          .getProjections
//          .asScala
//          .map(p => p -> r.getValue(p))
//          .toMap
//      )
//
//    }else if(r instanceof QuerySolution) {
//
//      this (
//        r
//          .varNames()
//          .asScala
//          .map(p => p -> r.get(p))
//          .toMap
//      )
//
//    }
//    else
//      r = null
//  }

}